src/main/java/io/redlink/more/data/service/GarminService.java

```diff
@@ -136,10 +139,34 @@ public boolean updateUserPermissions(String userId, List<String> permissions) {
         if (participants.isEmpty()) {
             return false;
         }
-        participants.forEach(participant -> participantKeyValueRepository.upsert(participant.studyId(), participant.participantId(), userId, Map.of(USER_ID_TYPE_KEY, "garmin", USER_PERMISSIONS_KEY, permissions)));
+        participants.forEach(participant ->
+                participantKeyValueRepository.upsert(
+                        participant.studyId(),
+                        participant.participantId(),
+                        userId,
+                        Map.of(USER_ID_TYPE_KEY, GARMIN_KEY_TYPE, USER_PERMISSIONS_KEY, permissions)
+                )
+        );
         return true;
     }
 
+    public void deregisterParticipant(Long studyId, int participantId) {
+        try {
+            var data = getUserAccessData(studyId, participantId);
+            if (data.isPresent()) {
+                sendDeregistration(data.get());
+                deleteUserAccessToken(studyId, participantId);
+            }
+
+            var participantWithKeyValue = keyValuesForParticipant(studyId, participantId);
+            participantWithKeyValue.forEach(participantKeyValue ->
+                    deleteGarminUserId(studyId, participantId, participantKeyValue.key())
+            );
+        } catch (RuntimeException e) {
+            LOG.error("Could not deregister Garmin User:", e);
```

Comment:
```suggestion
            LOG.error("Could not deregister Garmin User for participant {} of study {}:", participantId, studyId, e);
```
Im besten Fall kann man auch noch den `participantKeyValue.key()` loggen, damit man dann evtl. manuell aufräumen kann. Sind ja keine Personenbezogenen Daten.

Reply:
ich bin noch immer der Meinung, dass man keine `RuntimeException` abfangen sollte, sondern lieber akzeptiert, dass eine Exception fliegt. Wenn dann würde ich nur in methoden ein ExceptionHandling machen die externe Services aufrufen - also wohl in `deleteGarminUserId(..)`. Wenn die Datenbankabfragen zum laden der `participantWithKeyValue` fehlschlagen, dann ist sowieso mehr im argen.

Reply:
ahh ... das `deleteGarminUserId` ist auch nur eine DB Operation ... dann würde ich auf alle Fälle kein Exception Handling in dieser Methode machen